cilium-cli/status/k8s.go

```diff
@@ -308,6 +309,23 @@ func (k *K8sStatusCollector) podStatus(ctx context.Context, status *Status, name
 	return nil
 }
 
+func (k *K8sStatusCollector) ciliumNodeAnnotations(ctx context.Context, status *Status) error {
+	nodes, err := k.client.ListCiliumNodes(ctx)
```

Comment:
Instead of pulling the whole CiliumNode we could pull only the `PartialObjectMetadata` if we set the `Accept` header accordingly (like in `pkg/k8s/synced/crd.go`). It would make this significantly more complicated, but would save a bit of time and transmitted bytes. Not sure if it's worth it.

Reply:
Was curious, so implemented it:
```

func (c *Client) ListCiliumNodeAnnotations(ctx context.Context) (map[string]map[string]string, error) {
	const pomListHeader = "application/json;as=PartialObjectMetadataList;v=v1;g=meta.k8s.io,application/json;as=PartialObjectMetadataList;v=v1beta1;g=meta.k8s.io,application/json"
	rest := c.CiliumClientset.CiliumV2().RESTClient()
	opts := metav1.ListOptions{}
	list := metav1.PartialObjectMetadataList{}
	err := rest.Get().
		Name("ciliumnodes").
		SetHeader("Accept", pomListHeader).
		VersionedParams(&opts, metav1.ParameterCodec).
		Do(context.TODO()).
		Into(&list)
	if err != nil {
		return nil, err
	}
	annotations := map[string]map[string]string{}
	for _, meta := range list.Items {
		annotations[meta.Name] = meta.Annotations
	}
	return annotations, nil
}
```

yay or nay?